rskj-core/src/test/java/co/rsk/peg/bitcoin/BitcoinTestUtils.java

```diff
@@ -48,4 +56,36 @@ public static Sha256Hash createHash(int nHash) {
 
         return Sha256Hash.wrap(bytes);
     }
+
+    public static List<BtcECKey.ECDSASignature> extractSignaturesFromTxInput(TransactionInput txInput) {
+        String btcHexSignaturesPrefix = "30";
+        Script scriptSig = txInput.getScriptSig();
+        List<ScriptChunk> chunks = scriptSig.getChunks();
+        Script redeemScript = new Script(chunks.get(chunks.size() - 1).data);
+        RedeemScriptParser parser = RedeemScriptParserFactory.get(redeemScript.getChunks());
+
+        // This is the last chunk after the signatures.
+        int lastSigChunk;
+        RedeemScriptParser.MultiSigType multiSigType = parser.getMultiSigType();
+
+        if (multiSigType == RedeemScriptParser.MultiSigType.STANDARD_MULTISIG
+                || multiSigType == RedeemScriptParser.MultiSigType.FAST_BRIDGE_MULTISIG
+        ) {
+            lastSigChunk = chunks.size() - 1;
+        } else {
+            lastSigChunk = chunks.size() - 2;
+        }
+
+        List<BtcECKey.ECDSASignature> signatures = new ArrayList<>();
+        for (int i = 1; i < lastSigChunk; i++) {
+            ScriptChunk chunk = chunks.get(i);
+            if (!chunk.isOpCode() && chunk.data.length > 0) {
+                // Bitcoin signatures always starts with the prefix ‘30’.
```

Comment:
I think the same. I just added the if checking the prefix as a double check. And yes, according to the source I researched, that prefix is for all networks. Here is one of the sources: https://github.com/libbitcoin/libbitcoin-system/wiki/ECDSA-and-DER-Signatures.

But since it's not mandatory and it is enough with the above if, I'll remove it.

Reply:
Removed.